src/main/java/com/myFirstProject/myFirstProject/service/BasketServiceImpl.java

```diff
@@ -34,23 +38,46 @@
     @Autowired
     private ReqConverterService reqConverterService;
 
+    @Autowired
+    private PromoCodeRepository promoCodeRepository;
```

Comment:
lets inject beans via constructor instead of direct field injection

Reply:
See this: 
https://stackoverflow.com/questions/40620000/spring-autowire-on-properties-vs-constructor

Reply:
+